File: mpp/lab7/server/Repository/RepositoryError.cs
using System;
using System.Runtime.Serialization;

namespace server.Repository
{
    [Serializable]
    internal class RepositoryError : Exception
    {
        public RepositoryError()
        {
        }

        public RepositoryError(string message) : base(message)
        {
        }

        public RepositoryError(string message, Exception innerException) : base(message, innerException)
        {
        }
    }
}

File: mpp/lab8/server/src/main/java/repository/RepositoryError.java
package repository;

import com.fasterxml.jackson.annotation.JsonAutoDetect;

import java.io.Serializable;

public class RepositoryError extends Exception {
    RepositoryError(String message) {
        super(message);
    }

    public RepositoryError(Exception ex) {
        super(ex);
    }

    public Document getDocument() {
        return new Document(this.getMessage());
    }

    public class Document implements Serializable {
        private final boolean error = true;
        private final String message;

        private Document(String message) {
            this.message = message;
        }


        public boolean isError() {
            return error;
        }

        public String getMessage() {
            return message;
        }
    }
}
